src/upgrades.ts

```diff
@@ -149,7 +153,11 @@ async function upgradeProxy<T extends Contract>(
     opts?.proxyOpts
   )) as T
 
-  const deploymentTransaction = newContractInstance.deploymentTransaction()
+  // This is a workaround to get the deployment transaction. The upgradeProxy attaches
+  // the deployment transaction to the field under a different name than ethers
+  // contract.deploymentTransaction() function expects.
+  const deploymentTransaction =
+    newContractInstance.deployTransaction as unknown as ContractTransactionResponse
```

Comment:
Does it mean it is an issue in the OpenZeppelin upgrade plugin? If so, we should open an issue in their GitHub repository and link it from here.

Reply:
Probably it is something on their side unless that's again an issue with changes between different major ethers versions. Will check what is going on, if `ethers` was ever using that field name

Reply:
yeah so maybe someone forgot to fix that - in ethers v5 there is indeed `deployTransaction` ([docs](https://docs.ethers.org/v5/api/contract/example/#example-erc-20-contract--deploying-a-contract)) but OpenZeppelin's hardhat-upgrades package is already on ethers v6 ([link](https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/%40openzeppelin/hardhat-upgrades%403.0.1/packages/plugin-hardhat/package.json#L52))